config.h

```diff
@@ -4,44 +4,172 @@
 
 #include <map>
 #include <string>
+#include <optional>
+#include <vector>
+#include "resourcew.h"
+#include <set>
 
-enum ConfigOption {
-	_CONFIG_OPTIONS_START = 1000,
-	YonkStepSize,
-	YonkHomeDrift,
-	YonkEmotionScale,
-	TimeDivisor,
-	MaxColors,
-	SpriteCount,
-	SpriteSize,
-	YonkShakeFactor,
-	YonkPattern,
-	PatternChangeInterval,
-	IsPatternFixed,
-	ImpostorChance,
-	YonkPalette,
-	PlayOverDesktop,
-	_CONFIG_OPTIONS_END
-};
 
-const static std::map<int, std::wstring> config_names = {
-	{ YonkStepSize, L"YonkStepSize" },
-	{ YonkHomeDrift, L"YonkHomeDrift" },
-	{ YonkEmotionScale, L"YonkEmotionScale" },
-	{ TimeDivisor, L"TimeDivisor" },
-	{ MaxColors, L"MaxColors" },
-	{ SpriteCount, L"SpriteCount" },
-	{ SpriteSize, L"SpriteSize" },
-	{ YonkShakeFactor, L"YonkShakeFactor" },
-	{ YonkPattern, L"YonkPattern" },
-	{ PatternChangeInterval, L"PatternChangeInterval" },
-	{ IsPatternFixed, L"IsPatternFixed" },
-	{ ImpostorChance, L"ImpostorChance" },
-	{ YonkPalette, L"YonkPalette" },
-	{ PlayOverDesktop, L"PlayOverDesktop" },
+struct Cfg {
```

Comment:
I just cannot seem to decide on what the name of this type should be. I've decided on something very short since it's used all over the place, and I'm not happy with this but I'm OK with it.